builtin/providers/rundeck/resource_job.go

```diff
@@ -0,0 +1,556 @@
+package rundeck
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/apparentlymart/go-rundeck-api/rundeck"
+)
+
+func resourceRundeckJob() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateJob,
+		Update: UpdateJob,
+		Delete: DeleteJob,
+		Exists: JobExists,
+		Read:   ReadJob,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"project_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"log_level": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "INFO",
+			},
+
+			"allow_concurrent_executions": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"max_thread_count": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+
+			"continue_on_error": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"rank_order": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "ascending",
+			},
+
+			"rank_attribute": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"preserve_options_order": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"command_ordering_strategy": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "node-first",
+			},
+
+			"node_filter_query": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"node_filter_exclude_precedence": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"option": &schema.Schema{
+				Type:     schema.TypeList,
```

Comment:
Rundeck is kinda weird in that the order is *optionally* significant: it's used if the ``preserve_options_order`` is set, but ignored if it's not.